tests/jballerina-unit-test/src/test/resources/test-src/bir/bir-dump/cacheInserts

```diff
@@ -8,57 +8,57 @@ public cacheInserts function() -> () {
+    %9(SYNTHETIC) ballerina/cache:2.0.0:Cache;
     %13(TEMP) ();
-    %15(LOCAL) ();
-    %16(LOCAL) error{map<anydata | readonly>} | ();
+    %15(SYNTHETIC) ();
+    %16(SYNTHETIC) error{map<anydata | readonly>} | ();
     %18(TEMP) string;
     %19(TEMP) any;
     %20(TEMP) string;
     %23(TEMP) ();
     %24(TEMP) boolean;
-    %25(LOCAL) ();
-    %30(LOCAL) error;
-    %35(LOCAL) ();
-    %36(LOCAL) ballerina/cache:2.0.0:CacheError | ();
-    %45(LOCAL) ();
-    %50(LOCAL) error;
-    %55(LOCAL) ();
-    %56(LOCAL) ballerina/cache:2.0.0:CacheError | ();
-    %65(LOCAL) ();
-    %70(LOCAL) error;
-    %75(LOCAL) ();
-    %76(LOCAL) ballerina/cache:2.0.0:CacheError | ();
-    %85(LOCAL) ();
-    %90(LOCAL) error;
-    %95(LOCAL) ();
-    %96(LOCAL) ballerina/cache:2.0.0:CacheError | ();
-    %105(LOCAL) ();
-    %110(LOCAL) error;
-    %115(LOCAL) ();
-    %116(LOCAL) ballerina/cache:2.0.0:CacheError | ();
-    %125(LOCAL) ();
-    %130(LOCAL) error;
-    %135(LOCAL) ();
-    %136(LOCAL) ballerina/cache:2.0.0:CacheError | ();
-    %145(LOCAL) ();
-    %150(LOCAL) error;
-    %155(LOCAL) ();
-    %156(LOCAL) ballerina/cache:2.0.0:CacheError | ();
-    %165(LOCAL) ();
-    %170(LOCAL) error;
-    %175(LOCAL) ();
-    %176(LOCAL) ballerina/cache:2.0.0:CacheError | ();
-    %185(LOCAL) ();
-    %190(LOCAL) error;
-    %195(LOCAL) ();
-    %196(LOCAL) ballerina/cache:2.0.0:CacheError | ();
-    %205(LOCAL) ();
-    %210(LOCAL) error;
-    %215(LOCAL) ();
-    %216(LOCAL) ballerina/cache:2.0.0:CacheError | ();
-    %225(LOCAL) ();
-    %230(LOCAL) error;
+    %25(SYNTHETIC) ();
+    %30(SYNTHETIC) error;
+    %35(SYNTHETIC) ();
+    %36(SYNTHETIC) ballerina/cache:2.0.0:CacheError | ();
+    %45(SYNTHETIC) ();
+    %50(SYNTHETIC) error;
+    %55(SYNTHETIC) ();
+    %56(SYNTHETIC) ballerina/cache:2.0.0:CacheError | ();
+    %65(SYNTHETIC) ();
+    %70(SYNTHETIC) error;
+    %75(SYNTHETIC) ();
+    %76(SYNTHETIC) ballerina/cache:2.0.0:CacheError | ();
+    %85(SYNTHETIC) ();
+    %90(SYNTHETIC) error;
+    %95(SYNTHETIC) ();
+    %96(SYNTHETIC) ballerina/cache:2.0.0:CacheError | ();
+    %105(SYNTHETIC) ();
+    %110(SYNTHETIC) error;
+    %115(SYNTHETIC) ();
+    %116(SYNTHETIC) ballerina/cache:2.0.0:CacheError | ();
+    %125(SYNTHETIC) ();
+    %130(SYNTHETIC) error;
+    %135(SYNTHETIC) ();
+    %136(SYNTHETIC) ballerina/cache:2.0.0:CacheError | ();
+    %145(SYNTHETIC) ();
+    %150(SYNTHETIC) error;
+    %155(SYNTHETIC) ();
+    %156(SYNTHETIC) ballerina/cache:2.0.0:CacheError | ();
+    %165(SYNTHETIC) ();
+    %170(SYNTHETIC) error;
+    %175(SYNTHETIC) ();
+    %176(SYNTHETIC) ballerina/cache:2.0.0:CacheError | ();
+    %185(SYNTHETIC) ();
+    %190(SYNTHETIC) error;
+    %195(SYNTHETIC) ();
+    %196(SYNTHETIC) ballerina/cache:2.0.0:CacheError | ();
+    %205(SYNTHETIC) ();
+    %210(SYNTHETIC) error;
+    %215(SYNTHETIC) ();
+    %216(SYNTHETIC) ballerina/cache:2.0.0:CacheError | ();
+    %225(SYNTHETIC) ();
+    %230(SYNTHETIC) error;
```

Comment:
Just for my clarity what does the new `SYNTHETIC` type mean? I think I have explicitly checked for `LOCAL` types and temp types when reusing variables during optimization. Using a new type might break all those code. There might be other places as well. Shall we also mention in the PR description on the need to introduce this new type as it appears to be a big change

Reply:
Synthetic variables are the generated variables which we add in the desugar phase. These aint temporary variables w.r.t CBE and not LOCAL variables w.r.t CFE. Hence the new type